### PR TITLE
Update emotion colors for themes

### PR DESCRIPTION
**Pull request by noranda**
_Note: This branch is based off `nb/storybook-docs-ezapplayout` and should therefore be merged after https://github.com/ezcater/recipe/pull/955._

## What did we change?
- [add new colors to ezColors and update ezFulfillmentTheme and ezMarketplaceTheme to new colors](https://github.com/ezcater/recipe/commit/31d3d64d7626845c5b22ca011e468b353d689176) 
- [update welcome page button and toc link colors to common.green100](https://github.com/ezcater/recipe/commit/809362af1e020ea33ea4030a0d44424717b479d1)
- [update migration guide to note property color token changes and add color swatches](https://github.com/ezcater/recipe/commit/f7654404dc89eb13ac85b47b3c7c99f249b4f225) 

## Why are we doing this?
This is part of an effort to switch our doc site from Gatbsy to Storybook Docs.

Based on [design](https://www.figma.com/file/Ko43jjjwHKzT6pYSqU0Hzz/MUI-Foundations-(test)?type=design&node-id=34-2891&t=jNA8wvSU78r4bGqW-0).

## Screenshot(s) / Gif(s):
Screenshot 2023-05-16 at 9 58 06 AM
Screenshot 2023-05-16 at 9 58 21 AM
Screenshot 2023-05-16 at 9 57 28 AM

## Checklist

- [x] Create a changeset (`yarn changeset`) with a summary of the changes

### CODE REVIEW

File: packages/recipe/docs/components/TableOfContents/TableOfContents.styles.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@ const isActiveLiStyles = {
 };
 
 const tocLinkStyles = {
-  color: ezTheme.palette.primary.main,
+  color: ezTheme.palette.secondary.main,
```
Comment (agrobbin) [inline packages/recipe/docs/components/TableOfContents/TableOfContents.styles.ts:16]: Seems like these should be fixed in #951 (specifically in https://github.com/ezcater/recipe/pull/951/commits/83102fc2fde221c8c04b050e8b92c61ffabd2eed).

File: packages/recipe/docs/guides/migration/MigrationV18.mdx [hunk 2]
```diff
@@ -57,6 +59,57 @@ If using any of these colors directly, they will need to be updated to the new c
 <EzButton color="common.green100">Button</EzButton>
 ```
 
+### 2. Theme property color tokens have changed for `ezTheme`, `ezMarketplaceTheme`, and `ezFulfillmentTheme`
+
+Design tokens for the following theme properties have changed. See new colors in our [theming page](?path=/docs/guides-theming--docs#colors). No syntax changes required when migrating, but be aware that the colors will change where used. New colors were also added to `ezColors` and are available using `theme.palette.common[colorName]`.
+
+#### `ezTheme` and `ezMarketPlaceTheme`
```
Comment (agrobbin) [inline packages/recipe/docs/guides/migration/MigrationV18.mdx:66]: Really minor, but I think this should be:

```suggestion
#### `ezTheme` and `ezMarketplaceTheme`
```

File: packages/recipe/src/themes/mui/ezTheme.ts [hunk 3]
```diff
@@ -117,7 +117,7 @@ const palette: EzPaletteOptions = {
     main: ezPalette.grey140,
     light: ezPalette.white,
     dark: ezPalette.black,
-    contrastText: ezPalette.black,
+    contrastText: ezPalette.white,
```
Comment (agrobbin) [inline packages/recipe/src/themes/mui/ezTheme.ts:120]: Having no context into the reasons behind this change, I'd love to understand why we're switching from black to white! If it's substantial enough, I wonder if this particular change should be in its own commit, with an explanation in the commit message (it seems somewhat distinct from the rest of the changes in this commit).

File: packages/recipe/docs/components/TableOfContents/TableOfContents.styles.ts (hunk 1 above)
Comment (noranda) [reply packages/recipe/docs/components/TableOfContents/TableOfContents.styles.ts:16]: I'm going to leave this as-is due to the amount of effort/rebasing it takes to change commits that are on chained branches like these are. I will try to put fixes in the correct commits going forward!

File: packages/recipe/src/themes/mui/ezTheme.ts (hunk 3 above)
Comment (noranda) [reply packages/recipe/src/themes/mui/ezTheme.ts:120]: This was actually incorrectly changed in https://github.com/ezcater/recipe/pull/956/commits/19831292928974981d2cd5713cf72578a9237e99 and then fixed back in https://github.com/ezcater/recipe/pull/956/commits/9c7789761e85b300606c738b1cb98fba06816f69. I can try to fix the commit histories here.

File: packages/recipe/src/themes/mui/ezTheme.ts [hunk 4]
```diff
@@ -165,8 +165,8 @@ const typography = {
 
 const getStatusChipStyles = themeProp => {
   return {
-    backgroundColor: palette[themeProp].light,
-    color: palette[themeProp].main,
+    backgroundColor: themeProp === 'neutral' ? palette.common.grey120 : palette[themeProp].light,
```
Comment (agrobbin) [inline packages/recipe/src/themes/mui/ezTheme.ts:168]: Something about this feels a little off to me. What's stopping us from setting `palette.neutral.light = palette.common.grey120` to avoid this ternary statement?

File: packages/recipe/docs/components/TableOfContents/TableOfContents.styles.ts (hunk 1 above)
Comment (agrobbin) [reply packages/recipe/docs/components/TableOfContents/TableOfContents.styles.ts:16]: Yep, no problem!

File: packages/recipe/src/themes/mui/ezTheme.ts (hunk 4 above)
Comment (noranda) [reply packages/recipe/src/themes/mui/ezTheme.ts:168]: This PR changes `palette.neutral.light` to now be white (designer request), so the background on the status chip doesn't show up on white backgrounds. Since we want the background to be visible on white background, we've changed it to be a light gray instead.
Screenshot 2023-05-23 at 4 26 32 PM

File: packages/recipe/src/themes/mui/ezTheme.ts (hunk 3 above)
Comment (noranda) [reply packages/recipe/src/themes/mui/ezTheme.ts:120]: Fixed!

File: packages/recipe/src/themes/mui/ezTheme.ts (hunk 4 above)
Comment (agrobbin) [reply packages/recipe/src/themes/mui/ezTheme.ts:168]: OK I understand. I'm still a little iffy about the custom override here though. I'm cool with it in this PR, but IMO, even having a palette color key that was `lightButNotWhite` (or something similar) seem less brittle than an override like this deep in a single component style function.

**Review by agrobbin — APPROVED** [review]
